hugr-py/src/hugr/_ops.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Generic, Protocol, TypeVar, TYPE_CHECKING
+from hugr.serialization.ops import BaseOp
+import hugr.serialization.ops as sops
+import hugr.serialization.tys as tys
+
+if TYPE_CHECKING:
+    from hugr._hugr import Hugr, Node, Wire
+
+
+class Op(Protocol):
+    @property
+    def num_out(self) -> int | None:
```

Comment:
it can't be known in general - but can be for many of those! (which I've added). Am unsure about Custom but I've added it for now. It should be left as the default `None` for SerWrap.